### PR TITLE
feat(NODE-6086): add Double.fromString() method

### PR DESCRIPTION
**Pull request by aditi-khare-mongoDB**
### Description
Add static `Double.fromString()` method.

##### Is there new documentation needed for these changes?
Yes, there are new API docs.

#### What is the motivation for this change?
NODE-3660 user ticket, our validation in the Double is lacking. In V7, we will add `Double.fromString(value)` validation call into the `Double` constructor's `string` case.

### Add static `Double.fromString` method
This method attempts to create an `Double` type from a string, and will throw a `BSONError` on any string input that is not representable as a `IEEE-754 64-bit double`.
 Notably, this method will also throw on the following string formats:
   - Strings in non-decimal formats (exponent notation, binary, hex, or octal digits)
   - Strings with characters other than sign, numeric, floating point, or slash characters (Note: `'Infinity'`, `'-Infinity'`, and `'NaN'` input strings are still allowed)
   - Strings with leading and/or trailing whitespace
Strings with leading zeros, however, are also allowed.

### Double check the following

- [ ] Ran `npm run check:lint` script
- [ ] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [ ] PR title follows the [correct format](https://www.conventionalcommits.org/en/v1.0.0/): `type(NODE-xxxx)[!]: description`
  - Example: `feat(NODE-1234)!: rewriting everything in coffeescript`
- [ ] Changes are covered by tests
- [ ] New TODOs have a related JIRA ticket

### CODE REVIEW

**Comment by billouboq** [comment]
@nbbeeken Sorry to bother you in this PR, this feels like the only way of communicating since there is no issue in this repository 😆 
And sorry Aditi to pollute your PR with this comment 🙏 

I saw that there is a branch for date optimisations here : NODE-5955-improve-date

Is it something that will get released at some point ?

File: src/double.ts [hunk 1]
```diff
@@ -32,6 +33,34 @@ export class Double extends BSONValue {
...
+   * Notably, this method will also throw on the following string formats:
+   * - Strings in non-decimal formats (exponent notation, binary, hex, or octal digits)
+   * - Strings with characters other than numeric, floating point, or leading sign characters (Note: 'Infinity', '-Infinity', and 'NaN' input strings are still allowed)
+   * - Strings with leading and/or trailing whitespace
+   *
+   * Strings with leading zeros, however, are also allowed
+   *
+   * @param value - the string we want to represent as an double.
+   */
+  static fromString(value: string): Double {
```
Comment (baileympearson) [inline src/double.ts:49]: same comment here as on the `Int32.fromString()` PR: can we throw descriptive errors for each case we're checking for in the `if` statement?

File: test/node/double.test.ts [hunk 2]
```diff
@@ -225,6 +225,59 @@ describe('BSON Double Precision', function () {
...
+        ['leading and trailing whitespace', '    89   ', 89],
+        ['fake positive infinity', '2e308'],
+        ['fake negative infinity', '-2e308'],
+        ['fraction', '3/4'],
+        ['foo', 'foo']
+      ];
+
+      for (const [testName, value, expectedDouble] of acceptedInputs) {
+        context(`when case is ${testName}`, () => {
+          it(`should return Double that matches expected value`, () => {
```
Comment (baileympearson) [inline test/node/double.test.ts:261]: Same comment here about test titles.

File: src/double.ts (hunk 1 above)
Comment (aditi-khare-mongoDB) [reply src/double.ts:49]: Yep, on it

File: test/node/double.test.ts (hunk 2 above)
Comment (aditi-khare-mongoDB) [reply test/node/double.test.ts:261]: Just changed them, going to push after my tests run!

**Review by baileympearson — APPROVED** [review]

**Pull request merged**